consensus/src/user/provisioners.rs

```diff
@@ -136,6 +136,28 @@ impl Provisioners {
         self.members.insert(pubkey_bls, stake)
     }
 
+    /// Subtract `amount` from a staker, returning the stake left
+    ///
+    /// Return None if the entry was not found or `amount` is higher than
+    /// current stake
```

Comment:
Is it smart to have the same return value for a non-existing stake and overflowing subtraction? I would think returning a `Result` might make sense here, but I don't have a good overview on how and where this function will be used.

Reply:
It's not smart I agree
However, I thought that introducing an error for distinguish the cases was a bit overkilling for our need.

Indeed both cases (entry not found or amount higher), identify a bug in our code and the remediation is to resync the provisioners from scratch asking them to the stake contract with

```rust
            if let Err(e) = selective_update {
                warn!("Resync provisioners due to {e:?}");
                let state_hash = blk.header().state_hash;
                let new_prov = vm.get_provisioners(state_hash)?;
                provisioners_list.update_and_swap(new_prov)
            }
```

Having the "bug" logger somewhere is enough IMO